lib/gcov/include/uk/gcov.h

```diff
@@ -0,0 +1,30 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2023, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+#ifndef __UK_GCOV_H__
+#define __UK_GCOV_H__
+
+/* The linker script supplies the start and end symbols.
+ * We employ array syntax to circumvent potential
+ * small-data area problems
+ */
+extern const struct gcov_info *const __gcov_info_start[];
+extern const struct gcov_info *const __gcov_info_end[];
+
+/*
+ * Buffer size used to store the coverage information in memory
+ * that doubles every time the buffer is full. Adjust based on
+ * memory constraints or if you know the resulted file size
+ * in order to avoid unnecessary reallocations.
+ */
+static const size_t INITIAL_BUFFER_SIZE = 100 * (1 << 10); /* 100 KiB */
```

Comment:
By convention we use `#define` for these. Saying that, this should probably be provided as a Kconfig option instead 🤔